src/Language/Haskell/LSP/VFS.hs

```diff
@@ -163,7 +163,9 @@ changeFromServerVFS initVfs (J.RequestMessage _ _ _ params) = do
 -- ---------------------------------------------------------------------
 virtualFileName :: FilePath -> J.NormalizedUri -> VirtualFile -> FilePath
 virtualFileName prefix uri (VirtualFile _ file_ver _) =
-  prefix </> show (hash (J.fromNormalizedUri uri)) ++ "-" ++ show file_ver ++ ".hs"
+  let uri_raw = J.fromNormalizedUri uri
+      basename = maybe "" takeFileName (J.uriToFilePath uri_raw)
+  in prefix </> show (hash uri_raw) ++ "-" ++ show file_ver ++ "-" ++ basename ++ ".hs"
```

Comment:
Thanks, but a minor nit.  If I am viewing the temp directory sorted by filename, different versions of the same file are going to be scattered all over the place. It may be best to put the hash last, just before the extension.

And for bonus points left pad the version number with zeros, e.g. `001`.